File: 953.go
import "slices"

func isAlienSorted(words []string, order string) bool {
	alphabet := make(map[byte]byte, len(order))
	for i, c := range order {
		alphabet[byte(c)] = byte(i)
	}

	for i, s := range words {
		word := make([]byte, len(s))
		for i, c := range s {
			word[i] = 'a' + alphabet[byte(c)]
		}
		words[i] = string(word)
	}

	return slices.IsSorted(words)
}

File: 953.rs
impl Solution {
    pub fn is_alien_sorted(mut words: Vec<String>, order: String) -> bool {
        let mut alphabet: std::collections::HashMap<char,_> = std::collections::HashMap::with_capacity(order.len());
        let mut idx = 0;
        for c in order.chars() {
            alphabet.insert(c, idx);
            idx += 1;
        }

        let mut new_words = Vec::with_capacity(words.len());
        for s in words.iter() {
            let mut new_s = String::with_capacity(s.len());
            for c in s.chars() {
                if let Some(h) = alphabet.get(&c) {
                    unsafe {
                        new_s.push(char::from_u32_unchecked(97 + *h));
                    }
                }
            }

            new_words.push(new_s);
        }

        new_words.is_sorted()
    }
}
